core/src/main/scala/dagr/core/config/Configuration.scala

```diff
@@ -27,32 +27,79 @@ import java.io.File
 import java.nio.file.{Files, Path, Paths}
 import java.time.Duration
 
+import com.typesafe.config.{ConfigParseOptions, ConfigFactory, Config}
 import com.typesafe.config.ConfigException.Generic
 import dagr.core.execsystem.{Cores, Memory}
 import dagr.core.util.{Io, LazyLogging}
 
+import scala.collection.SortedSet
 import scala.reflect.runtime.universe._
 
 /**
   * Companion object to the Configuration trait that keeps track of all configuration keys
   * that have been requested so that they can be reported later if desired.
   */
-private[core] object Configuration {
-  private val RequestedKeys = collection.mutable.TreeSet[String]()
+private[core] object Configuration extends ConfigurationLike {
+  // A sorted set tracking all the configuration keys that are requested
+  private[config] val RequestedKeys = collection.mutable.TreeSet[String]()
+
+  // The global configuration instance for dagr
+  private[config] var _config: Config = ConfigFactory.load()
+
+  /** Implement the abstract method from BaseConfiguration to return the actual config instance. */
+  override private[config] def config = _config
+
+  // Keys for configuration values used in dagr core
+  object Keys {
+    val CommandLineName = "dagr.command-line-name"
+    val ScriptDirectory = "dagr.script-directory"
+    val LogDirectory = "dagr.log-directory"
+    val SystemCores = "dagr.system-cores"
+    val SystemMemory = "dagr.system-memory"
```

Comment:
I didn't even notice - freaking IntelliJ and it's need to mess with my whitespace on paste.